rs/https_outcalls/consensus/src/pool_manager.rs

```diff
@@ -291,9 +291,21 @@ impl CanisterHttpPoolManagerImpl {
             return Vec::new();
         };
 
+        let existing_signed_requests: BTreeSet<_> = canister_http_pool
+            .get_validated_shares()
+            .map(|share| (share.signature.signer, share.content.id))
```

Comment:
You need to check the registry version too.
If the registry version changes, we need to generate new shares, with the same `(signer, request_id)` pair, just for a different `registry_version`. So this needs to be a triple.

Otherwise, we will timeout requests, that happened before a registry version change, since we will not re-generate their shares.

Reply:
Didn't we find out that shares aren't regenerated anyway if the registry version changes?

Reply:
If this is the case, we may want to change that at some point, so I still think it makes sense to keep the registry version in the map.

Reply:
Added the registry version, please TAL.

Reply:
What would determine if we may want to change that at some point? In this case we will have to adjust the theoretic bound and introduce some more checks. Otherwise someone could send us an arbitrary amount of shares, each with a different registry versions

Reply:
Btw @Sawchord notified me via Slack, that the suggestion is indeed incorrect and we don't need the registry version here, that's why I removed it again, yesterday.